pkg/controller/daemon/update.go

```diff
@@ -246,7 +246,7 @@ func (dsc *DaemonSetsController) controlledHistories(ds *apps.DaemonSet) ([]*app
 
 	// List all histories to include those that don't match the selector anymore
 	// but have a ControllerRef pointing to the controller.
-	histories, err := dsc.historyLister.List(labels.Everything())
```

Comment:
This doesn't list history in all namespaces. The informer is already namespace scoped: https://github.com/kubernetes/client-go/blob/611184f7c43ae2d520727f01d49620c7ed33412d/informers/apps/v1/controllerrevision.go#L54-L77

Reply:
I appreciate your pointing out my mistake